src/pdf/filter.rs

```diff
@@ -0,0 +1,143 @@
+use mupdf_sys::*;
+
+use std::ffi::{CStr, CString};
+use std::os::raw::{c_char, c_void};
+use std::ptr;
+
+use crate::pdf::PdfDocument;
+use crate::{context, Image, Matrix, Rect};
+
+#[derive(Clone, Copy)]
+pub struct PdfFilterOptions {
+    pub(crate) inner: *mut pdf_filter_options,
+}
+
+// Callback types
+// TODO: I don't think this is necessary for anything.
+pub type ImageFilter = fn(ctm: &Matrix, name: &str, image: &Image) -> Image;
+pub type TextFilter = fn(ucsbuf: i32, ucslen: i32, trm: &Matrix, ctm: &Matrix, bbox: &Rect) -> i32;
+pub type AfterTextObject = fn(doc: &PdfDocument, chain: &pdf_processor, ctm: &Matrix);
+pub type EndPage = fn();
+
+impl PdfFilterOptions {
+    pub(crate) unsafe fn from_raw(ptr: *mut pdf_filter_options) -> Self {
+        Self { inner: ptr }
+    }
+
+    pub fn ascii(&self) -> bool {
+        unsafe { (*self.inner).ascii != 0 }
+    }
+
+    pub fn set_ascii(&mut self, value: bool) -> &mut Self {
+        unsafe {
+            (*self.inner).ascii = if value { 1 } else { 0 };
+        }
+        self
+    }
+
+    pub fn recurse(&self) -> bool {
+        unsafe { (*self.inner).recurse != 0 }
+    }
+
+    pub fn set_recurse(&mut self, value: bool) -> &mut Self {
+        unsafe {
+            (*self.inner).recurse = if value { 1 } else { 0 };
+        }
+        self
+    }
+
+    pub fn sanitize(&self) -> bool {
+        unsafe { (*self.inner).sanitize != 0 }
+    }
+
+    pub fn set_sanitize(&mut self, value: bool) -> &mut Self {
+        unsafe {
+            (*self.inner).sanitize = if value { 1 } else { 0 };
+        }
+        self
+    }
+
+    pub fn instance_forms(&self) -> bool {
+        unsafe { (*self.inner).instance_forms != 0 }
+    }
+
+    pub fn set_instance_forms(&mut self, value: bool) -> &mut Self {
+        unsafe {
+            (*self.inner).instance_forms = if value { 1 } else { 0 };
+        }
+        self
+    }
+
+    pub fn image_filter(&self) -> Option<impl Fn(&Matrix, &str, &Image) -> Image> {
```

Comment:
Let's keep things simple and remove these getters, they are not particularly useful.

Reply:
Agreed. It was just for consistency, if anything.